hawkular-wildfly-monitor/src/main/java/org/hawkular/agent/monitor/service/MonitorService.java

```diff
@@ -819,4 +832,85 @@ private void connectToFeedCommChannel() throws Exception {
         feedComm = new FeedComm(this.httpClientBuilder, this.configuration, this.feedId, this.dmrServerInventories);
         feedComm.connect();
     }
+
+    private static final class ServerAddressProvider implements InventoryStorage {
+        private final InventoryStorage delegate;
+        private final ManagedServer server;
+
+        public ServerAddressProvider(ManagedServer server, InventoryStorage delegate) throws UnknownHostException {
+            super();
+            this.delegate = delegate;
+            this.server = server;
+        }
+
+        private InetAddress[] resolveHost() throws UnknownHostException {
+            String host = null;
+            if (server instanceof RemoteDMRManagedServer) {
+                RemoteDMRManagedServer remoteServer = (RemoteDMRManagedServer) server;
+                host = remoteServer.getHost();
+            } else if (server instanceof LocalDMRManagedServer) {
+                host = InetAddress.getLocalHost().getCanonicalHostName();
+            } else {
+                throw new IllegalStateException("Unexpected type of managed server '" + server.getClass().getName()
+                        + "'; expected '" + RemoteDMRManagedServer.class.getName() + "' or '"
+                        + LocalDMRManagedServer.class.getName() + "'. Please report this bug.");
+            }
+            MsgLogger.LOG.tracef("About to resolve IP addresses for host [%s]", host);
+            return InetAddress.getAllByName(host);
+        }
+
+        @Override
+        public void storeResourceType(ResourceType<?, ?, ?, ?> resourceType) {
+            delegate.storeResourceType(resourceType);
+        }
+
+        @Override
+        public void storeResource(Resource<?, ?, ?, ?, ?> resource) {
+            final String WF_TYPE_NAME = "WildFly Server";
+            final String IP_ADDRESSES_PROPERTY_NAME = "IP Addresses";
+            String typeName = resource.getResourceType().getName().getNameString();
+            if (WF_TYPE_NAME.equals(typeName) && resource instanceof DMRResource) {
+                DMRResource dmrResource = (DMRResource) resource;
+                /* modify the config so that only the bound addresses are added */
+                DMRResourceConfigurationPropertyInstance adrProp = null;
+                for (DMRResourceConfigurationPropertyInstance prop : dmrResource.getConfigurationProperties()) {
+                    if (IP_ADDRESSES_PROPERTY_NAME.equals(prop.getName().getNameString())) {
```

Comment:
How is that checking for a/the public interface?
Suppose you have a dual-homed AS with the public (http) interface on 192.168.1.2 and the privat (management) interface on 10.1.2.3 then we want 192.168.1.2 returned and not the 10.* address